dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/ProgramRuleVariableObjectBundleHook.java

```diff
@@ -94,16 +96,40 @@
     private void validateUniqueProgramRuleName( ObjectBundle bundle,
         ProgramRuleVariable programRuleVariable, Consumer<ErrorReport> addReports )
     {
-        Query<ProgramRuleVariable> query = getProgramRuleVariableQuery( programRuleVariable );
+        List<ProgramRuleVariable> prvWithSameNameAndSameProgram = getProgramRuleVariableQuery( programRuleVariable )
+            .getResultList();
 
-        int allowedCount = UPDATE_STRATEGIES.contains( bundle.getImportMode() ) ? 1 : 0;
+        // When no PRV with same name and program, validation passes
+        if ( prvWithSameNameAndSameProgram.isEmpty() )
+        {
+            return;
+        }
 
-        if ( query.getResultList().size() > allowedCount )
+        // since prvWithSameNameAndSameProgram is not empty, fail if it's an
+        // insert or if there is more than one prv with same name and same
+        // program.
+        if ( !UPDATE_STRATEGIES.contains( bundle.getImportMode() ) || prvWithSameNameAndSameProgram.size() > 1 )
```

Comment:
If there is more than one in the DB already, doesn't mean that data in DB is wrong?

Reply:
that's true, but this specific case means that if INSERT, in DB it is expected to be 0, if UPDATE, in DB there can be at most 1.
when in database there are 2 or more, the check will fail anyway. Or am I missing something ?

Reply:
ok maybe I got it... maybe we can have a failing "legal" update just because data in DB is bad. Yes this might happen as well, but it shouldn't honestly. Here I just wanted to make it simpler the following part by assuming that the list only contained one element.

Reply:
then what I can do is remove size()>1 condition and since it's an update, ensure at least one in the List have the same UID as the PRV we are processing in the validator, what would you think ?